### DISCUSSION TITLE
Server only discoverable on port 2457

### DISCUSSION THREAD

**Discussion by nminchow** [lloesche/valheim-server-docker#32] [Q&A] [answered]
When I rely on the default env var (or manually supply `SERVER_PORT: 2456`), the server seems to only be discoverable on steam at port 2457. My server isn't showing up in the server lists either, and I suspect the issues are related. This may not be a problem with this image, but I had noticed other users experiencing the same issue:

https://github.com/lloesche/valheim-server-docker/issues/26#issuecomment-779185280

**Comment by InB4DevOps** [comment]
The server will listen on SERVER_PORT + 1

The server not showing up in the server list is not related to the port.

**Comment by nminchow** [comment]
> The server will listen on SERVER_PORT + 1

Is this just a quirk of the executable? I spent nearly 2 hours trying to debug why I couldn't connect while specifying an IP.

If it is a quirk of the exe, we can close this issue. I'm just curious if this is something with the dockerfile or a product of the executable itself.

**Comment by lloesche** [comment, marked as answer]
The Steam server browser query port and the game server run on two different ports. The later by default on 2456 the former on gameserver_port+1 so usually 2457. It's just how the devs coded it.
That behavior is documented here https://github.com/lloesche/valheim-server-docker#steam-server-favorites--lan-play

**Reply by nminchow** [reply]
The docs are what sent me down the correct path eventually - so that is appreciated! I just didn't even think to look in the image's readme when I was initially debugging.

**Reply by nodu** [reply]
"The server not showing up in the server list is not related to the port." How do we solve this issue then? Cheers!

**Reply by InB4DevOps** [reply]
> "The server not showing up in the server list is not related to the port." How do we solve this issue then? Cheers!

I'm afraid there's nothing we as server admins can do about that. 
With SO many servers and the steam server list currently being limited to ~5000 results we'd need some kind of server regions so that you could look in the server log which region it's registered in and then tell your players "look in region xy" in order to find the server. 

This would of course require changes to Steam's server list management and the game itself.

**Reply by InB4DevOps** [reply]
BTW: when you can connect to your server via IP:port then most likely it can be discovered in the server list - you just happen to be in the wrong region. I had a friend of mine from Switzerland connect to a nearby VPN and only then he was able to find my server (which is hosted in Germany). There are just too many servers at the moment.

**Reply by nodu** [reply]
Thanks for the info! 
Is this the same as the in-game search function as well? Players needing to connect via the Steam server list is a huge barrier of entry for new joiners. Having the in-game filtering work would be excellent.
